### DISCUSSION TITLE
GLIBC_2.33 not found

### DISCUSSION THREAD

**Discussion by dpo** [Homebrew/discussions#3996] [Linux] [answered]
Hi. I'm having the following error when building on a `ubuntu-latest` GitHub Actions VM:
```
 /usr/bin/ld: /home/linuxbrew/.linuxbrew/Cellar/gcc/12.2.0/bin/../libexec/gcc/x86_64-pc-linux-gnu/12/liblto_plugin.so: error loading plugin: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /home/linuxbrew/.linuxbrew/Cellar/gcc/12.2.0/bin/../libexec/gcc/x86_64-pc-linux-gnu/12/liblto_plugin.so)
```
The formula I'm building installs the latest `gcc`. What is the resolution of the above error?

Link to build log: https://github.com/coin-or-tools/homebrew-coinor/actions/runs/3575600426/jobs/6013598246

Thanks in advance!

**Comment by Bo98** [comment, marked as answer]
I'll look into this. But a fix will be to use `ubuntu-22.04` instead of `ubuntu-latest` (though `ubuntu-latest` should point to 22.04 soon: https://github.com/actions/runner-images/issues/6399)

**Reply by dpo** [reply]
That solves the issue! Many thanks!

**Comment by williscool** [comment]
So sorting this out was absolute hell for me using homebrew on wsl windows

I had to

1. update wsl `wsl --upgrade`
2. upgrade to ubuntu 22.04
3. reinstall glibc
4. rebuild openssl from source
5. install x86_64-linux-gnu-binutils
6. had to figure out which of 3 or 4 permutations of 4 magic incantations (setting CPPFLAGS, LDFLAGS, PKG_CONFIG_PATH, and changing the PATH variable to put various things first).... turns out I needed to have `export PATH="/home/linuxbrew/.linuxbrew/opt/x86_64-linux-gnu-binutils/bin:$PATH"` first in the path

just to get crystal to build properly again


also the added bonus of trying to figure out why `GLIBC_2.33` or `GLIBC_2.34` can't be found even though I `brew --link`ed and rebuilt libraies until I could barely think.... 

all because an auto update started breaking everything when I just wanted to update `git`... which didn't even work :(

P.S. also it is IMPOSSIBLE to google this stuff `ld` `cc` the flags you are supposed to set... and where? evn vars? the command line? do you put them in quotes or not?

You fine folks that work on this stuff for homebrew all the time are heroes. I would LOSE MY MIND.

Anyway thanks for reading my rant that cost me more than I would like to admit over the past while to sort out

moral of the story I will be adding 

```
HOMEBREW_NO_AUTO_UPDATE=1 
```

to my bashrc from now on
